sql/core/src/main/scala/org/apache/spark/sql/functions.scala

```diff
@@ -102,6 +102,27 @@ object functions {
     Column(literalExpr)
   }
 
+  /**
+   * Creates a [[Column]] of literal value.
+   *
+   * The passed in object is returned directly if it is already a [[Column]].
+   * If the object is a Scala Symbol, it is converted into a [[Column]] also.
+   * Otherwise, a new [[Column]] is created to represent the literal value.
+   *
+   * @group normal_funcs
+   * @since 2.2.0
+   */
+  def typedLit[T : TypeTag](literal: T): Column = {
+    literal match {
```

Comment:
This match statement is slightly hair raising (I know this is copied from `lit(...), how about:
```scala
literal match {
  case c: Column => c
  case s: Symbol => new ColumnName(s.name)
  case _ => Column(Literal.create(literal))
}
```

You could also consider mapping the untyped `lit(..)` function to this function.